Made use love.handlers.

gphg opened the pull request:
Enchanted and working version of: https://github.com/gphg/scenery/commit/c77bb2d04697b27370ca27a8e6abfb15a335811b

* Utilize `love.handlers`. Not listed and not re-added callbacks are:  `errhand`, `errorhandler`.

* Added `Scenary:hook`, which inject all scene callbacks into targeted table: `love`. Keep your `main.lua` clean.

* `setScene` is now not a global function. It can be accessed via `self.setScene`.

gphg committed:
Made use love.handlers.

Enchanted and working version of: https://github.com/gphg/scenery/commit/c77bb2d04697b27370ca27a8e6abfb15a335811b

* Utilize `love.handlers`. Not listed and not re-added callbacks are:  `errhand`, `errorhandler`.

* Added `Scenary.hook`, which inject all scene callbacks into targeted table: `love`. Keep your `main.lua` clean.

* `setScene` is now not a global function. It can be accessed via `self.setScene`.

gphg commented:
I didn't change the files under `examples` directory nor the README.md. If this PR get merged, please change these.

Changes on this PR are:

* `loveCallbacks` is pulled from `love.handlers`. 
  * Related: #3
* `setScene` is now can be accessed on every scene. It is not a global function.
* added `hook`, here's the example:
```lua
-- main.lua
local SceneryInit = require("scenery")
local scenery = SceneryInit("scene1", "states")

scenery:hook(love)
```
```lua
-- states/scene1.lua
local scene1 = {}

function scene1:load()
    self.counter = 0
end

function scene1:draw()
    love.graphics.print(self.counter .. " people love Scenery", 300, 250)
end

function scene1:update()
    self.counter = self.counter + 1

    if self.counter > 500 then
        self.setScene("scene2", self.counter)
    end
end

return scene1
```
```lua
-- states/scene2.lua
local scene2 = {}

function scene2:load(number)
    print("More than " .. number .. " people love Scenery.")
end

function scene2:keypressed(key)
    if key == 'return' then
        self.setScene('scene1')
    end
end

return scene2
```

gphg commented:
This has bug which cause `load` get called twice. To be fixed, do not merge this yet.

gphg committed:
Prevent `load` get called twice.

paltze commented:
OK, I will edit the README and examples accordingly and merge this soon

gphg commented:
This PR is ready. 👍

paltze commented:
There is some sort of problem with the new `scenery.lua` for me. Have you tested it? Trying the new `scenery.lua` in a working project I got a black window. So I tested it as such:
`main.lua`
```lua
local SceneryInit = require('scenery')
local scenery = SceneryInit('test')
scenery:hook(love)

```
`scenes/test.lua`
```lua
local test = {}

function test:load()
    print('Hello Load')
end

function test:update()
    print('Hello Update')
end

return test
```
I only get `Hello Update` in the console

gphg commented:
It has to do with recently removed `load` on https://github.com/paltze/scenery/pull/4/commits/5a900b6f2e7715e68b9538110f0648f03c23d5ce. When I put it back, `load` get called twice.

Try to revert it, if console print `Hello Load` only once, then it must have to do with my side.

gphg commented:
You are right. It has to do with my side. On my project, I call `scenery.setScene('test', arg)` after the hook. I have to do this because there is no way on passing argument on default scene. This makes `load` on test scene get called twice.

gphg committed:
Fix: `load` not getting called.

Revert previous commit: https://github.com/paltze/scenery/pull/4/commits/5a900b6f2e7715e68b9538110f0648f03c23d5ce

gphg commented:
This completely my opinion. I think the example of manual definition callback (before `hook` is implemented) on README.md should be keep, even if it's just a bare minimum information.

This might including one of example too. For wider.possible and historical reason.

paltze commented:
Maybe you are right, I'll add it
